src/manager/node.c

```diff
@@ -260,53 +276,97 @@ static int node_match_unit_properties_changed(sd_bus_message *m, void *userdata,
+                LIST_FOREACH_SAFE(subs, usub, next_usub, usubs->subs) {
+                        Subscription *sub = usub->sub;
+                        int r = sub->handle_unit_property_changed(sub->monitor, node->name, unit, interface, m);
+                        if (r < 0) {
+                                hirte_log_error("Failed to emit UnitPropertyChanged signal");
+                        }
+                }
         }
 
-        UnitSubscription *usub = NULL;
-        UnitSubscription *next_usub = NULL;
-        LIST_FOREACH_SAFE(subs, usub, next_usub, usubs->subs) {
-                Subscription *sub = usub->sub;
-                int r = sub->handle_unit_property_changed(sub->monitor, node->name, sub->unit, interface, m);
-                if (r < 0) {
-                        hirte_log_error("Failed to emit UnitPropertyChanged signal");
+        const UnitSubscriptionsKey wildcard_key = { (char *) SYMBOL_WILDCARD };
+        UnitSubscriptions *usubs_wildcard = hashmap_get(node->unit_subscriptions, &wildcard_key);
+        if (usubs_wildcard != NULL) {
+                UnitSubscription *usub = NULL;
+                UnitSubscription *next_usub = NULL;
+                LIST_FOREACH_SAFE(subs, usub, next_usub, usubs_wildcard->subs) {
+                        Subscription *sub = usub->sub;
+
+                        /* only emit signal once per monitor */
+                        if (usubs != NULL &&
+                            unit_subscription_contains_monitor(usubs->subs, (Monitor *) sub->monitor)) {
+                                continue;
+                        }
+
+                        int r = sub->handle_unit_property_changed(sub->monitor, node->name, unit, interface, m);
+                        if (r < 0) {
+                                hirte_log_error("Failed to emit UnitPropertyChanged signal");
+                        }
                 }
         }
 
         return 1;
 }
 
+
 static int node_match_unit_new(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *error) {
         Node *node = userdata;
         const char *unit = NULL;
         const char *reason = NULL;
 
         int r = sd_bus_message_read(m, "ss", &unit, &reason);
         if (r < 0) {
-                hirte_log_error("Invalid UnitNew signal");
+                hirte_log_errorf("Invalid UnitNew signal: %s", strerror(-r));
                 return 0;
         }
 
         const UnitSubscriptionsKey key = { (char *) unit };
-        UnitSubscriptions *usubs = NULL;
-
-        usubs = hashmap_get(node->unit_subscriptions, &key);
-        if (usubs == NULL) {
-                return 0;
+        UnitSubscriptions *usubs = hashmap_get(node->unit_subscriptions, &key);
+        if (usubs != NULL) {
+                usubs->loaded = true;
+                if (is_wildcard(unit)) {
+                        usubs->active_state = UNIT_ACTIVE;
+                        free(usubs->substate);
+                        usubs->substate = strdup("running");
```

Comment:
This initializes the virtual substate for a wildcard to "running", so maybe we should use that in the agent too?